warehouse/upload.go

```diff
@@ -2077,3 +2083,42 @@ func (job *UploadJobT) GetLocalSchema() warehouseutils.SchemaT {
 func (job *UploadJobT) UpdateLocalSchema(schema warehouseutils.SchemaT) error {
 	return job.schemaHandle.updateLocalSchema(schema)
 }
+
+func (job *UploadJobT) RefreshPartitions(loadFileStartID, loadFileEndID int64) error {
+	if !slices.Contains(warehouseutils.TimeWindowDestinations, job.upload.DestinationType) {
+		return nil
+	}
+
+	var (
+		repository schemarepository.SchemaRepository
+		err        error
+	)
+
+	if repository, err = schemarepository.NewSchemaRepository(job.warehouse, job); err != nil {
+		return fmt.Errorf("create schema repository: %w", err)
+	}
+
+	// Refresh partitions if exists
+	for tableName := range job.upload.UploadSchema {
+		loadFiles := job.GetLoadFilesMetadata(warehouseutils.GetLoadFilesOptionsT{
+			Table:   tableName,
+			StartID: loadFileStartID,
+			EndID:   loadFileEndID,
+		})
+
+		// This is best done every 100 files, since it's a batch request for updates in Glue
+		partitionBatchSize := config.GetInt("Warehouse.refreshPartitionBatchSize", 99)
+		for i := 0; i < len(loadFiles); i += partitionBatchSize {
+			end := i + partitionBatchSize
+			if end > len(loadFiles) {
+				end = len(loadFiles)
+			}
+
```

Comment:
Would it be better to move the batching logic to the `GlueSchemaRepository` or GlueClient ?

It seems destination-specific logic.

Also why `GetInt("Warehouse.refreshPartitionBatchSize", 99)` is not a 100 ?

Reply:
There is a limit set by the SDK for the batching to be 100. 
https://github.com/rudderlabs/rudder-server/pull/1603/commits/5c8a8c9e7b8af30c6bee14c4c2d534f5a5861cb4#